control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1183,3 +1213,255 @@ func (r *reconciler) reconcileAWSIdentityWebhook(ctx context.Context) []error {
 
 	return errs
 }
+
+func (r *reconciler) destroyCloudResources(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Ensuring resource creation is blocked in cluster")
+	if err := r.ensureResourceCreationIsBlocked(ctx); err != nil {
+		return err
+	}
+	log.Info("Ensuring image registry storage is removed")
+	removed, err := r.ensureImageRegistryStorageRemoved(ctx)
+	if err != nil || !removed {
```

Comment:
Wouldn't it be better to not block cleanup of other resource type by the previous resource type and only return prior to the status condition setup if there are any remaining resources?